### PR TITLE
LT-20698: Use non-cultured date formats

### PR DESCRIPTION
**Pull request by papeh**
Write dates using a culture-neutral format ensures they are consistent and anyone can read them.

Part of https://jira.sil.org/browse/LT-20698

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/248)

### CODE REVIEW

File: tests/SIL.LCModel.Tests/DomainImpl/NotebookTests.cs [hunk 1]
```diff
@@ -56,6 +55,10 @@ public void SubrecordOfKey()
 			var subFish10 = MakeSubRecord(recordFish, "Cleaning the fish", 10);
 			Assert.That(subFish2.SubrecordOfKey(false), Is.LessThan(subFish10.SubrecordOfKey(false)));
 		}
+
+		//[Test]
+		//public void ShortNameTSS_
```
Comment (jasonleenaylor) [inline tests/SIL.LCModel.Tests/DomainImpl/NotebookTests.cs:60]: ?

**Review by jasonleenaylor — APPROVED** [review]
Commented out code can be removed, otherwise looking good.

File: src/SIL.LCModel/DomainImpl/ReadWriteServices.cs [hunk 2]
```diff
@@ -128,30 +130,42 @@ internal static Guid LoadGuid(XElement reader)
 		/// </summary>
 		internal static DateTime LoadDateTime(XElement reader)
 		{
-			if (reader == null) throw new ArgumentNullException("reader");
+			if (reader == null) throw new ArgumentNullException(nameof(reader));
 
-			// ENHANCE: it would be better to use DateTime.Parse instead of parsing this ourselves.
```
Comment (papeh) [inline src/SIL.LCModel/DomainImpl/ReadWriteServices.cs:133]: can't use DateTime.Parse with the European date format dd.mm.yyyy (not sure if any projects have this problem)

**Comment by papeh** [comment]
Needs a changelog entry: fixed migration, added DateTime extension

**Pull request merged**